test/app/features/profile/data/repositories/user_profile_repository_test.dart

```diff
@@ -63,22 +78,25 @@ void main() {
         await sut.anonymousMode(toggle: true);
         // assert
         verify(
-          apiProvider.post(
+          () => apiProvider.post(
             path: endPoint,
             parameters: parameters,
           ),
-        );
-      });
-      test('should receive ValidField', () async {
-        // arrange
-        _setUpMockPost();
-        final actual = right(const ValidField());
-        const toggle = false;
-        // act
-        final matcher = await sut.anonymousMode(toggle: toggle);
-        // assert
-        expect(actual, matcher);
+        ).called(1);
       });
+      test(
+        'receive ValidField',
+        () async {
+          // arrange
+          _setUpMockPost();
+          final actual = right(const ValidField());
+          const toggle = false;
+          // act
+          final matcher = await sut.anonymousMode(toggle: toggle);
+          // assert
+          expect(actual, matcher);
```

Comment:
Sei que no caso foi mantida a estrutura dos testes, porém poderia aproveitar e ajustar o nome das variáveis? Pois caso o teste falhe, pode ficar um pouco confuso de se entender a mensagem sendo que apareceria erroneamente que o retorno de `anonymousMode` é o esperado.

```suggestion
          _setUpMockPost();
          final expected = right(const ValidField());
          const toggle = false;
          // act
          final actual = await sut.anonymousMode(toggle: toggle);
          // assert
          expect(actual, expected);
```